core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/aggregator/Aggregator.java

```diff
@@ -0,0 +1,297 @@
+/**
+ * 
+ */
+package edu.uci.ics.texera.dataflow.aggregator;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+
+import edu.uci.ics.texera.api.constants.ErrorMessages;
+import edu.uci.ics.texera.api.dataflow.IOperator;
+import edu.uci.ics.texera.api.exception.TexeraException;
+import edu.uci.ics.texera.api.field.DateField;
+import edu.uci.ics.texera.api.field.DoubleField;
+import edu.uci.ics.texera.api.field.IntegerField;
+import edu.uci.ics.texera.api.field.StringField;
+import edu.uci.ics.texera.api.field.TextField;
+import edu.uci.ics.texera.api.field.IField;
+import edu.uci.ics.texera.api.schema.Attribute;
+import edu.uci.ics.texera.api.schema.AttributeType;
+import edu.uci.ics.texera.api.schema.Schema;
+import edu.uci.ics.texera.api.schema.Schema.Builder;
+import edu.uci.ics.texera.api.tuple.Tuple;
+import edu.uci.ics.texera.dataflow.common.AbstractSingleInputOperator;
+
+/**
+ * @author avinash
+ * 
+ * Aggregator operator is used to perform aggregation (like sum, count, min, max, average) operations on a column.
+ *
+ */
+public class Aggregator extends AbstractSingleInputOperator
```

Comment:
In general there's a bit lack of comments in this class. I suggest to add some more comments, since it's the "main" class.